Sources/OpenFeature/Provider/FeatureProvider.swift

```diff
@@ -6,10 +6,12 @@ public protocol FeatureProvider: EventPublisher {
     var metadata: ProviderMetadata { get }
 
     /// Called by OpenFeatureAPI whenever the new Provider is registered
-    func initialize(initialContext: EvaluationContext?)
+    /// This must throw in case of error, using OpenFeature errors whenever possible
+    /// It is expected that the implementer is slow and blocking (e.g. network), the caller must deal with wrapping this into a background Task if necessary
+    func initialize(initialContext: EvaluationContext?) throws
```

Comment:
I think we need docs like this for user-facing `setContext` and `setContextAndWait`, as well, so it's clear that the former can take time and should be called from the UI thread.